corehq/apps/app_manager/models.py

```diff
@@ -352,7 +362,28 @@ def get_property_names(self):
 
     @property
     def is_subcase(self):
-        return self.parent_tag
+        return bool(self.parents)
+
+    @classmethod
+    def wrap(cls, data):
+        _assert = soft_assert(to=['nhooper@dimagi.com'])
+        if not _assert('parent_tag' not in data,
+                       'AdvancedAction not migrated. Old application build may have been reverted.'):
+            if data['parent_tag']:
+                # Lazy-migrate single parent index to list of parent/host indices
+                parent = {
+                    'tag': data['parent_tag'],
+                    'reference_id': data.get('parent_reference_id', 'parent'),
+                    'relationship': data.get('relationship', 'child')
+                }
+                if hasattr(data.get('parents'), 'append'):
+                    data['parents'].append(parent)
+                else:
+                    data['parents'] = [parent]
+                del data['parent_tag']
+                data.pop('parent_reference_id', None)
```

Comment:
Its good to clean up else next time it will still be there.

On 5 August 2015 at 10:24, Norman Hooper <notifications@github.com> wrote:

> In corehq/apps/app_manager/models.py
> <https://github.com/dimagi/commcare-hq/pull/7749#discussion_r36279166>:
>
> > +        _assert = soft_assert(to=['nhooper@dimagi.com'])
> > +        if not _assert('parent_tag' not in data,
> > +                       'AdvancedAction not migrated. Old application build may have been reverted.'):
> > +            if data['parent_tag']:
> > +                # Lazy-migrate single parent index to list of parent/host indices
> > +                parent = {
> > +                    'tag': data['parent_tag'],
> > +                    'reference_id': data.get('parent_reference_id', 'parent'),
> > +                    'relationship': data.get('relationship', 'child')
> > +                }
> > +                if hasattr(data.get('parents'), 'append'):
> > +                    data['parents'].append(parent)
> > +                else:
> > +                    data['parents'] = [parent]
> > +                del data['parent_tag']
> > +                data.pop('parent_reference_id', None)
>
> Actually, do we need to bother removing unused keys from data? Or is it
> better/safer to clean up?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/7749/files#r36279166>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa